src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/02-spawner.py

```diff
@@ -73,6 +73,7 @@ def service_for_jhub_apps(name, url):
                 "external": True,
             },
             "oauth_no_confirm": True,
+            "oauth_redirect_uri": "/services/:name/oauth_callback",
```

Comment:
I suspect the fix here is to _remove_ the `oauth_no_confirm` here, not add an oauth_redirect_uri, because these don't look like these are applications that oauth with JupyterHub. Is that right? The reason JupyterHub 5 raises with this configuration is that some configuration suggests that OAuth is desired, but insufficient configuration is provided for it to work. So it's rightly raising, but perhaps not informatively enough.

Do you perhaps have a suggestion for the error message that might have helped you better here?

1. authenticate users with JupyterHub as OAuth provider, or
2. communicate with the JupyterHub API via tokens issued under these service names?

Or is this perhaps using services config as a way to generate some links to things that aren't really Services at all, from JupyterHub's perspective?

Reply:
Thank you for chiming in with helpful suggestions!

> these don't look like these are applications that oauth with JupyterHub

@aktech do you know if this is right? I suspect that jhub-apps might in fact be doing these things, but I am not quite sure. Do you think we should keep/remove `oauth_no_confirm`?

Reply:
Thanks again for the review @minrk 🎉 

>these don't look like these are applications that oauth with JupyterHub

Yes, they don't. They just use keycloak directly.

>Do you think we should keep/remove oauth_no_confirm?

Yes correct. I already removed them from example `jupyterhub_config.py` in jhub-apps: https://github.com/nebari-dev/jhub-apps/pull/220/files#diff-26e325937baab623ebecf944e3691cfc6d509853168cf154a403089ea6a93a1e